js/main.js

```diff
@@ -11,60 +27,127 @@ $photoURL.addEventListener('input', function (event) {
 
 var $form = document.querySelector('.newEntry');
 
+var currentEditing = false;
+
 $form.addEventListener('submit', function (event) {
   event.preventDefault();
 
   var newEntry = {
     title: $form.elements.title.value,
     photoURL: $form.elements.photoURL.value,
-    notes: $form.elements.notes.value,
-    ID: data.nextEntryId
+    notes: $form.elements.notes.value
   };
 
-  data.entries.unshift(newEntry);
-  data.nextEntryId++;
-
-  $form.reset();
-  $newEntryImg.setAttribute('src', 'images/placeholder-image-square.jpg');
+  if (currentEditing === false) {
+    newEntry.ID = data.nextEntryId;
+    data.entries.unshift(newEntry);
+    data.nextEntryId++;
+    $entryGallery.prepend(addEntry(data.entries[0]));
+  } else {
+    newEntry.ID = currentEditID;
+    for (var i = 0; i < data.entries.length; i++) {
+      // debugger;
+      if (data.entries[i].ID === currentEditID) {
+        data.entries[i] = newEntry;
+        var $allEntries = document.querySelectorAll('.uniqueEntry');
+        for (var v = 0; v < $allEntries.length; v++) {
+          if (parseInt($allEntries[v].getAttribute('data-entry-id')) === currentEditID) {
+            $allEntries[v].replaceWith(addEntry(newEntry));
+          }
+        }
+      }
+    }
+  }
 
+  data.editing = null;
+  resetForm();
   $noEntries.remove();
-  $entryGallery.prepend(addEntry(data.entries[0]));
-
   swapViews('entries');
+  currentEditing = false;
 });
 
+function resetForm() {
+  NewOrEdit.textContent = 'New Entry';
+  $form.reset();
+  $newEntryImg.setAttribute('src', 'images/placeholder-image-square.jpg');
+  $form.elements.title.removeAttribute('value');
+  $form.elements.photoURL.removeAttribute('value');
+  $form.elements.notes.textContent = '';
+}
+
 var $body = document.querySelector('body');
 var allViews = document.querySelectorAll('.view');
+var NewOrEdit = document.querySelector('.newOrEdit');
+var currentEditID = null;
 
 $body.addEventListener('click', clickHandler);
 function clickHandler(event) {
   if (!event.target.matches('.swap')) {
     return;
   }
   event.preventDefault();
+
+  if (event.target.matches('.editing')) {
+    currentEditID = parseInt(event.target.closest('li').getAttribute('data-entry-id'));
+    editEntryInStorage(currentEditID);
+    currentEditing = true;
```

Comment:
You can remove this variable by just checking if the value of `data.editing` is not `null`